Mathlib/Data/Nat/FinMulAntidiagonal.lean

```diff
@@ -0,0 +1,252 @@
+/-
+Copyright (c) 2024 Arend Mellendijk. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Arend Mellendijk
+-/
+
+import Mathlib.Algebra.Order.Antidiag.Pi
+import Mathlib.NumberTheory.ArithmeticFunction
+
+/-!
+# Sets of tuples with a fixed product
+
+This file defines the finite set of `d`-tuples of natural numbers with a fixed product `n` as
+`Nat.finMulAntidiagonal`.
+
+## Main Results
+* There are `d^(ω n)` ways to write `n` as a product of `d` natural numbers, when `n` is squarefree
+(`card_finMulAntidiagonal_of_squarefree`)
+-/
+
+open Finset
+open scoped BigOperators ArithmeticFunction
+namespace PNat
+
+/-- The set of divisors of a positive natural number.
+This is `Nat.divisorsAntidiagonal` without a special case for `n = 0`. -/
+def divisorsAntidiagonal (n : ℕ+) : Finset (ℕ+ × ℕ+) :=
```

Comment:
Changing `Nat.divisorsAntidiagonal` would be a pretty big refactor. As it stands, it's used all over the `ArithmeticFunction` API, and it is used together with `Nat.divisors` which we would probably want to change as well. I don't think this would be bad per se, but I am not certain that the resulting API would be much better. 

The problem is that `PNat` just isn't very well-behaved. It doesn't fit neatly in the algebraic hierarchy (it's not even a semiring), so using it as is would require expanding the `PNat` API quit a bit. A lot of tactics with special cases for `Nat` also don't support `PNat` properly. 

```lean
example (x y : ℕ+) (h : x < y) : x + 1 < y + 1 := by
  /- linarith failed to find a contradiction -/
  linarith
```